src/services/ShSimulatorService.tsx

```diff
@@ -447,12 +540,70 @@ const ShSimulation = ({ children }: GenericContextProviderProps) => {
 		[getJobStatus]
 	);
 
+	const findFirstEstimatorName = useCallback(
+		async (
+			jobStatus: JobStatusData,
+			inputs: JobInputs | undefined,
+			givenEstimatorName?: string
+		) => {
+			const firstEstimatorName = inputs?.input.inputJson?.scoringManager.outputs[0].name;
```

Comment:
you can make a query to input.metadata to see if the project is of "EDITOR" or "FILES" type.
if the project comes from "FILES" type then whole `scoringManager` part will be missing and makes no sense to query it.
see again my comment here: https://github.com/yaptide/ui/pull/1790#issuecomment-2486027308

Reply:
I would prefer to stick to one convention of checking names. It would be just unnecessary if statement. This approach is common among React code and is shorter

Reply:
Which convention do you have in mind?

Reply:
What about situation that project is of files type and by mistake has wrong scoring manager field ?

Reply:
What will happen in user uploaded simulation if user do not clean the project?

Reply:
Input metadata is more accurate source of truth than presence of data in scoring manager